### PR TITLE
Add WorkshopLocation model

### PR DESCRIPTION
**Pull request by PetrDlouhy**
* use `AbstractText` model for it
* make text field as `MarkdownField` using `django-markdown-app`
Addresses issue #39

### CODE REVIEW

File: api/models/workshop.py [hunk 1]
```diff
@@ -23,6 +25,10 @@ class Workshop(CapacityMixin, Base):
         WorkshopDifficulty,
         verbose_name=_("Difficulty"),
     )
+    location = models.ForeignKey(
```
Comment (just-paja) [inline api/models/workshop.py]: Workshop can be located in different spots over the years.

File: api/tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from model_mommy import mommy
+
+
+def gen_func():
```
Comment (just-paja) [inline api/tests/__init__.py]: Why do we need to test this?

File: api_textual/models/workshop.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+"""Import Django models."""
```
Comment (just-paja) [inline api_textual/models/workshop.py]: File name mismatch: `workshop.py` vs `WorkshopLocation`.

File: api/models/workshop.py (hunk 1 above)
Comment (just-paja) [inline api/models/workshop.py]: @PetrDlouhy, yes. Also, `Workshop` can already be reused over the years. This change breaks that. What do you mean a LOT of changes? Do you mean having this relation as many2many over a year? Well, I guess that's just insane.

Changing lector logically changes the workshop content as well.

Comment (PetrDlouhy) [inline api/models/workshop.py]: Do we need to keep track of workshop over the years? I would say yes. I think, that we need to say, what lector had a workshop last year. What was the participants of that workshop. What was the content.

If we reuse workshop for a new year we either lose all that information from last year or we need to keep track of all the changes.

If we keep track of all that information (as suggested by linking workshop to different location over the years), we would need to store all the old descriptions, lectors, participants and other. That is a LOT of changes.

I propose, we don't reuse `Workshop` in a new year. We could copy them, but the old `workshop`s should stay as record of the past year.

Comment (just-paja) [inline api/models/workshop.py]: @PetrDlouhy, so you propose we have same workshop locations for every year? Now that's insane.

Comment (PetrDlouhy) [inline api/models/workshop.py]: Why? Workshop locations usually doesn't change between years. The gym is the same gym the next year. The workshop could have same name, but the content, lector and participants can change significantly. Or those thinks can change only slightly every year, but at the end it is completely different workshop.

Maybe I am just missing something substantial about how you want to use it. But you should explain yourself a bit more than in one sentence.

Comment (just-paja) [inline api/models/workshop.py]: @PetrDlouhy, it is unsure where the festival will be held every year. Do you remember Improtřesk on Klamovka? Improtřesk in Lužánky (Brno)? Do you remember using the Milevsko Nursery School building in 2015? Workshop locations have a great potential to every year. 2017 will definitely have different room dispositions to 2016.

Have it your way. Right now, your generating overhead. I'll change it when I have time.

Comment (PetrDlouhy) [inline api/models/workshop.py]: No, you are right, that something needs to be changed in the current design. I suggest linking workshop to current year. Maybe also location.

What do you suggest?

Comment (just-paja) [inline api/models/workshop.py]: Let's keep it this way for the moment and change it when we re-examine our needs.

**Comment by just-paja** [comment]
@PetrDlouhy , please fix the conflict and let's merge this.

File: api_textual/models/workshop.py (hunk 3 above)
Comment (PetrDlouhy) [inline api_textual/models/workshop.py]: This file was there by mistake. I removed it.

**Pull request merged**